### ISSUE TITLE
How to deal with redundant AddImport calls?

### ISSUE THREAD

**Issue by mmanyin** [GEOS-ESM/MAPL#259]
There are Gridded Components, such as HEMCO and TR, that allow the user to configure multiple members within them (HEMCO has "instances";  TR has "species"). Each member can be configured independently of the others.  Within the SetServices routine of HEMCO or TR, there is a loop to deal with each member in turn, and import the necessary fields by calling MAPL_AddImportSpec.  A problem arises  when more than one member wants to access the same field from the same file, without duplication of effort.  **MAPL does not allow a Gridded Component to call MAPL_AddImportSpec more than once using the same field name.  I would request that this restriction be relaxed to allow identical calls, or at least that a new MAPL routine be provided which checks if an Import has already been added.**  Currently, a GridComp must work to maintain a list of its Imports to avoid duplicates, or the user must use aliases to force ExtData to read data multiple times; neither approach is optimal.

**Comment by tclune** [comment]
Probably needs to the latter option.  What would it mean if the two specs were not consistent?

**Comment by atrayano** [comment]
Adding a new MAPL routine to check is a given spec is already there seems a good idea, and should be relatively easy to implement. @mmanyin , thanks for suggesting it!

**Comment by mmanyin** [comment]
@tclune  If the two specs were not consistent, that would clearly be an error.

**Comment by mathomp4** [comment]
@mmanyin Is this still needed?

@atrayano @bena-nasa Has this been implemented?

**Comment by mmanyin** [comment]
Yes, there are still TR species which share common input datasets.  The work-around, as coded at the top level of TR is still in place, and I assume that it's still needed.  As I recall, the solution in HEMCO was to use distinct import names, which likely leads to multiple copies of the data in memory.

**Comment by mathomp4** [comment]
Reexamined on 2021-Apr-12 with @bena-nasa and @tclune

**Comment by mmanyin** [comment]
When this was re-examined last April, was there a consensus?  I would still (very much) support a routine as @atrayano recommended on Mar 18.

**Comment by tclune** [comment]
Mentioning to @bena-nasa too, as he's the one more likely to touch the logic in ExtData for this.

**Comment by mmanyin** [comment]
@bena-nasa Could we have a routine to inquire "is there already an import field with name xxx?" and could it please return attributes of the field as well.  TR can call this from SetServices and decide whether to AddImport.  Clearly we only want to add an import once for TR (even though it may be specified for multiple TR tracers).

**Comment by bena-nasa** [comment]
We should have this check for addimport/internal/export. That is reasonable. I will add it to the list.

**Comment by mmanyin** [comment]
@STALE-BOT  your setting is a bit aggressive!  I will add  "long term" tag.
